src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -190,7 +190,6 @@ internal static bool IsRecoverableException(Exception exception)
             // using 'GetType()' instead of 'is' as SecurityTokenUnableToValidException (and others) extend SecurityTokenInvalidSignatureException
             // we want to make sure that the clause for SecurityTokenUnableToValidateException is hit so that the ValidationFailure is checked
             return exception.GetType().Equals(typeof(SecurityTokenInvalidSignatureException))
-                   || exception is SecurityTokenInvalidSigningKeyException
```

Comment:
TBD: SecurityTokenInvalidSigningKeyException is recoverable exception or not.

Reply:
I don't think SecurityTokenInvalidSigningKeyException is valid.